pkg/client/fieldvalidation.go

```diff
@@ -0,0 +1,107 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package client
+
+import (
+	"context"
+
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+// WithStrictFieldValidation wraps a Client and configures strict field
+// validation, by default, for all write requests from this client. Users
+// can override the field validation for individual requests.
+func WithStrictFieldValidation(c Client) Client {
```

Comment:
If we make validation a parameter then this same func can be used for the other validation values as well

Reply:
Done in https://github.com/kubernetes-sigs/controller-runtime/pull/2860/commits/21366f20738216c29278e6b08a17c1c13116a533